### PR TITLE
Migrate to Wrangler 2

### PR DESCRIPTION
**Pull request by tytan652**
### Description

Wrangler (legacy) is depecated and no longer supported by CloudFlare.

worker crate has been updated to its latest version to enable Wrangler 2 support.

### Motivation and Context

Remake the worker usable with supported Wrangler version since I wanted to have a custom working for my projects.

### How Has This Been Tested?

I  published one and it seems to work.

### Types of changes

- Breaking change (fix or feature that would cause existing functionality to change)
  - No longer support Wrangler (legacy)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

**Comment by derrod** [comment]
The workflow would probably also need updating then?

**Comment by tytan652** [comment]
> The workflow would probably also need updating then?

I didn't see it, I did an update but not tested.

File: .github/workflows/cf_deploy.yaml [hunk 1]
```diff
@@ -24,11 +24,7 @@ jobs:
         uses: Swatinem/rust-cache@v1
 
       - name: Install wrangler
-        run: cargo install wrangler
-
-      - name: Build
-        if: github.event_name != 'workflow_dispatch'
-        run: wrangler build
```
Comment (derrod) [inline .github/workflows/cf_deploy.yaml:31]: The build-only version when pushing should not be removed

**Comment by derrod** [comment]
We'll see if it works I guess

**Comment by derrod** [comment]
Turns out the answer is no but the error makes no sense

File: .github/workflows/cf_deploy.yaml [hunk 2]
```diff
@@ -24,17 +24,17 @@ jobs:
         uses: Swatinem/rust-cache@v1
 
       - name: Install wrangler
-        run: cargo install wrangler
+        run: npm install -g wrangler
 
-      - name: Build
+       - name: Build
```
Comment (tytan652) [inline .github/workflows/cf_deploy.yaml:29]: The issue is here

**Pull request merged**